doc/PORTING.md

```diff
@@ -0,0 +1,141 @@
+# Porting How-To
+
+This document describes the requirements and necessary steps required to port
+`mcuboot` to a new target `OS`.
+
+# Requirements
+
+* `mcuboot` requires that the target provides a `flash` API with ability to
+  get flash alignment and read/write/erase individual sectors.
+
+* `mcuboot` doesn't bundle a cryptographic library, which means the target
+  OS must already have it bundled. The supported libraries at the moment are
+  `tinycrypt` and `mbed TLS`.
+
+* TODO: what more?
+
+# Steps to port
+
+## Configuration options macro
+
+Configuration options are checked with the macro `MYNEWT_VAL(x)` where `x`
+is the configuration to get the value for. This enables support for OSes with
+configuration systems that don't take the `#define ...` format. If the target
+OS uses `#define ...` one can simply use the following definition:
+
+```c
+#define MYNEWT_VAL(x) (x)
+```
+
+If your system uses another method, like build config files, a proper façade
+must be defined.
+
+## Main app and calling the bootloader
+
+From the perspective of the target OS, the bootloader can be seen as a library,
+so an entry point must be provided. This is likely a typical `app` for the
+target OS, and it must call the following function to run the bootloader:
+
+```c
+int boot_go(struct boot_rsp *rsp);
+```
+
+This function is located at `boot/bootutil/loader.c` and receives a `struct
+boot_rsp` pointer. The `struct boot_rsp` is defined as:
+
+```c
+struct boot_rsp {
+    /** A pointer to the header of the image to be executed. */
+    const struct image_header *br_hdr;
+
+    /**
+     * The flash offset of the image to execute.  Indicates the position of
+     * the image header.
+     */
+    uint8_t br_flash_id;
+    uint32_t br_image_addr;
+};
+```
+
+After running the management functions of the bootloader, `boot_go` returns
+an initialized `boot_rsp` which has pointers to the location of the image
+where the target firmware is located which can be used to jump to.
+
+## Flash access and flash Map
+
+* Regarding flash access the bootloader has two requirements:
+
+### hal_flash_align
+
+`mcuboot` needs to know the alignment of the flash. To get this information it
+calls `hal_flash_align`.
+
+TODO: this needs to die and move to flash_map...
+
+### flash_map
+
+The bootloader requires a `flash_map` to be able to know how the flash is
```

Comment:
Given that partition is the ordinary term for this, and map is kind of an unusual term, perhaps partition shouldn't be in quotes, and maybe map should be. Maybe: "The bootloader needs to now how the flash is partitioned, using a `flash_map`..."